spot-contracts/contracts/FeePolicy.sol

```diff
@@ -282,26 +282,35 @@ contract FeePolicy is IFeePolicy, OwnableUpgradeable {
             return (0, ONE);
         }
 
-        // Split the fees between the perp and vault systems
+        // When the system is over-subscribed, we charge no perp mint fee.
+        // The perp's share of the swap fee is zero and the vault gets the entire slice.
         uint256 totalFeePerc = vaultSwapFeePerc[
             keccak256(abi.encodePacked(address(vault.underlying()), address(perp)))
         ];
-        // For the fee split however, we still use the "pre"-minting or current state.
-        uint256 perpSharePerc = totalFeePerc.mulDiv(currentPerpTVL, currentPerpTVL + currentVaultTVL);
-        return (perpSharePerc, totalFeePerc - perpSharePerc);
+        return (0, totalFeePerc);
     }
 
     /// @inheritdoc IFeePolicy
     function computePerpToUnderlyingSwapFeePercs() external override returns (uint256, uint256) {
         // When user swaps perps for vault's underlying -> perps are redeemed by the vault
-        uint256 currentPerpTVL = perp.getTVL();
-        uint256 currentVaultTVL = vault.getTVL();
+        IFeePolicy.SubscriptionState memory s = computeSubscriptionState();
 
         // Split the fees between the perp and vault systems
         uint256 totalFeePerc = vaultSwapFeePerc[
             keccak256(abi.encodePacked(address(perp), address(vault.underlying())))
         ];
-        uint256 perpSharePerc = totalFeePerc.mulDiv(currentPerpTVL, currentPerpTVL + currentVaultTVL);
+        // When the system is over-subscribed, we charge no perp burn fee.
+        // The perp's share of the swap fee is zero and the vault gets the entire slice.
+        if (s.normalizedSubscriptionRatio > 1) {
+            return (0, totalFeePerc);
+        }
+
+        // When the system is under-subscribed, perp's share is computed as the minimum of
```

Comment:
Alternatively, we could give perp  `perpBurnFeePerc` and the vault `totalFeePerc-perpBurnFeePerc` provided `totalFeePerc >= perpBurnFeePerc`.